device_attributes/index.bs

```diff
@@ -0,0 +1,180 @@
+Repository: https://github.com/WICG/WebApiDevice
+Group: wicg
+</pre>
+
+<pre class=link-defaults>
+spec:html; type:interface; text:Navigator
+spec:url; type:dfn; text:origin
+spec:html; type:dfn; for:/; text:global object
+spec:webidl; type:dfn; text:resolve
+spec:webidl; type:dfn; text:record
+</pre>
+
+<pre class="anchors">
+
+</pre>
+
+# Model # {#model}
+
+## Managed devices ## {#managed-devices}
+
+The API is presumed to be used on devices which are not fully controlled by the end user, but rather by an external entity, the device administrator. Device administrators are given the power to fully control the managed devices.
+
+## Managed web applications ## {#managed-web-applications}
+
+The API is presumed to be used by web applications, which are understood and managed by the device administrators.
+
+## Device attributes ## {#device-attributes}
+
+The API is presumed to be used with a management infrastructure which allows administrators to manage a fleet of devices and configure properties for those devices.
+
+### Annotated Asset ID ### {#annotated-asset-id}
+
+An administrator-defined value which uniquely identifies a device within an organization.
+
+### Annotated Location ### {#annotated-location}
+
+An administrator-defined value which uniquely identifies a location within an organization.
+
+### Directory ID ### {#directory-id}
+
+An inventory management system-defined value which uniquely identifies a device within an organization.
+
+### Hostname ### {#hostname}
+
+An administrator-defined value which is used as the device hostname during DHCP request.
+
+### Serial Number ### {#serial-number}
+
+A manufacturer-defined value which uniquely identifies a device.
+
+# {{NavigatorManagedData}} interface # {#navigatormanageddata-interface}
+
+<xmp class="idl">
+[
+  SecureContext,
+  Exposed=Window
+] interface NavigatorManagedData : EventTarget {
+  // Device Attributes API.
+  Promise<DOMString> getAnnotatedAssetId();
+  Promise<DOMString> getAnnotatedLocation();
+  Promise<DOMString> getDirectoryId();
+  Promise<DOMString> getHostname();
+  Promise<DOMString> getSerialNumber();
+};
+</xmp>
+
+Methods on this interface typically complete asynchronously, queuing work on the <dfn>managed data task source</dfn>.
+
+## {{getAnnotatedAssetId()}} method ## {#getAnnotatedAssetId-method}
+
+  The <dfn method for="NavigatorManagedData">getAnnotatedAssetId()</dfn> method steps are:
+  1. Let |promise| be [=a new promise=].
+  1. Run the following steps in parallel:
+      1. If this API is not called by a managed web application, reject |promise| with a {{NotAllowedError}} DOMException.
+      1. If no [[#annotated-asset-id]] has been set by the administrator, resolve |promise| with {{undefined}} value.
```

Comment:
With the &lt;dfn> above you can do this:

```suggestion
      1. If no [=Annotated Asset ID=] has been set by the administrator, resolve |promise| with {{undefined}} value.
```

Same comment for the other attribute definition references.